src/foam/nanos/client/ClientBuilder.js

```diff
@@ -56,26 +57,33 @@ foam.CLASS({
             package: 'foam.nanos.client',
             name: 'Client',
             exports: [],
-            properties: [],
+            properties: []
           };
 
           var references = [];
 
           // Force hard reload when app version updates
-          self.nSpecDAO.find("appConfig").then(function(spec) {
-            var appConfig = spec.service;
-
-            client.exports.push(spec.name);
+          var appConfigService = self.ClientAppConfigService.create({
+            delegate: self.RetryBox.create({
+              maxAttempts: 2,
+              delegate: self.HTTPBox.create({
+                method: 'POST',
+                url: 'service/appConfigService'
+              })
+            })
+          });
+          var appConfigPromise = appConfigService.getAppConfig().then(function(appConfig) {
+            client.exports.push('appConfig');
             references = references.concat(foam.json.references(self.__context__, appConfig));
             client.properties.push({
-              name: spec.name,
+              name: 'appConfig',
               factory: function() {
-                return foam.json.parse(appConfig, null, this.__subContext__);
+                return appConfig.clone(this.__subContext__);
               }
             });
 
             var version   = appConfig.version;
-            if ( "CLIENT_VERSION" in localStorage ) {
+            if ( 'CLIENT_VERSION' in localStorage ) {
```

Comment:
NOTE: AppConfig version is now set during the running of Script 'main'.  The version is extracted from the jar manifest and set on AppConfig.

Reply:
Oh good to know. That shouldn't affect this code much though as that's the server setting up the app config that eventually makes its way to the client and will ultimately get checked here. Or is there something I'm missing I need to account for?

Reply:
Spoke with Joel offline. This is just good information to know but he also pointed out that groups have a getAppConfig that I'll need to be mindful of. Shouldn't affect this PR so I'm going to merge it now but I think we can use this infrastructure to clean some other things up.